java-tools/OsmAndServer/src/main/java/net/osmand/server/api/repo/PremiumUserFilesRepository.java

```diff
@@ -40,6 +44,14 @@ public interface PremiumUserFilesRepository extends JpaRepository<UserFile, Long
 			+ "WHERE uf.userid = :userid AND uf.name LIKE :folderName% AND uf.type = :type AND (uf.name, uf.updatetime) IN "
 			+ "(SELECT uft.name, MAX(uft.updatetime) FROM UserFile uft WHERE uft.userid = :userid GROUP BY uft.name)")
 	List<UserFile> findLatestFilesByFolderName(@Param("userid") int userid, @Param("folderName") String folderName, @Param("type") String type);
+
+	@Query(value = "SELECT u.id, u.userid, u.deviceid, u.type, u.name, u.updatetime, u.clienttime, u.filesize, u.zipfilesize, u.storage " +
+			"FROM user_files u " +
+			"WHERE u.userid = :userid " +
+			"AND u.type = 'GPX' " +
+			"AND u.quadtiles && :tiles " +
+			"ORDER BY u.updatetime DESC", nativeQuery = true)
+	List<Object[]> listGPXFilesByTiles(@Param(value = "userid") int userid, @Param(value = "tiles") String[] tiles);
```

Comment:
This doens't look good queries support normal UserFlile, you can return normal List<UserFile>